app/frontend/src/utils/permissionUtils.js

```diff
@@ -79,3 +79,14 @@ export async function determineFormNeedsAuth(formId, submissionId, next) {
   }
   next();
 }
+
+/**
+ * @function isFormPublic
+ * Returns true for a form object if this form can be viewed by public users
+ * @param {Object} form The form object fr
+ * @returns {boolean} TRUE if public
+ */
+export function isFormPublic(form) {
+  return form &&
+    (form.identityProviders && form.identityProviders.some(i => i.code === IdentityMode.PUBLIC));
```

Comment:
i forget if we're still attributing a public form submission to a user if they WERE logged in when they submitted the form. I think that was seen as a bug a few weeks back. because you could show 'my submissions' depending on if the user is logged in or not instead. just a thought. probably not worth changing now even of we wanted to.

Reply:
Yeah that's right, we _were_ doing that but we've fixed the bug. So a public form loses any knowledge of who submitted it before it hits the DB. The only link really is that you could have sent yourself an email with the link back.